### PR TITLE
[fortran/en] Adding a Fortran (95) Tutorial

### PR DESCRIPTION
**Pull request by robochat**
Here is a Fortran tutorial. I really don't know Fortran though so this was a learning exercise for me, it would be great if someone who does know it could check it over!

### CODE REVIEW

File: fortran95.html.markdown [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+filename: learnfortran.f95
+---
+
+Fortran is one of the oldest computer languages. It was developed in the 1950s 
+by IBM for numeric calculations (Fortran is an abreviation of "Formula 
+Translation"). Despite its age, it is still used for high-performance computing
+such as weather prediction. However, the language has changed considerably over
+the years, although mostly maintaining backwards compatibility; well known
+versions are FORTRAN 77, Fortran 90, Fortran 95, Fortran 2003, Fortran 2008 and
+Fortran2015.
```
Comment (vendethiel) [inline fortran95.html.markdown]: "Fortran 2015"

File: fortran95.html.markdown [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+    implicit none    !prevents dynamic declaration of variables (recommended!)
+    ! Implicit none must be redeclared in every function/program/module...
+    
+    ! IMPORTANT - Fortran is case insensitive.
+    real z
+    REAL Z2
+
+    real :: v,x    ! WARNING: default initial values are compiler dependent!    
+    real :: a = 3, b=2E12, c = 0.01
+    integer :: i,j,k=1,m
```
Comment (vendethiel) [inline fortran95.html.markdown]: Inconsistent spacing

File: fortran95.html.markdown [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+program example   !declare a program called example.
+
+    ! Indentation is not actually required but is recommended.
+
+    ! Declaring variables
+    ! ===================
+    
+    ! All declarations must come before statements and expressions.
+    
+    implicit none    !prevents dynamic declaration of variables (recommended!)
```
Comment (vendethiel) [inline fortran95.html.markdown]: Please space the comments' content

File: fortran95.html.markdown [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+    ! Assignment & Arithmetic
+    ! =======================
+
+    Z = 1    !assign to variable z declared above (case insensitive).
+    j = 10 + 2 - 3
+    a = 11.54  /  (2.3 * 3.1)
+    b = 2**3    !exponentiation
+    
+    !an aside:
+    print *, b    !print the variable 'b' to the command line (see I/O section).
```
Comment (vendethiel) [inline fortran95.html.markdown]: Why is this here?

File: fortran95.html.markdown [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+    ! Single-line if statement
+    if (z == a) b = 4  !condition always need surrounding parentheses.
+
+    if (z /= a) then !z not equal to a
+    ! Other symbolic comparisons are < > <= >= == /=
+      b = 4
+    else if (z .GT. a) then !z greater than a
+    ! Text equivalents to symbol operators are .LT. .GT. .LE. .GE. .EQ. .NE.  
+      b = 6
+    else if (z <  a) then !'then' must be on this line.
```
Comment (vendethiel) [inline fortran95.html.markdown]: Extra space before a

File: fortran95.html.markdown [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+        j=1
+      case (11:)    !all cases where i>=11
+        j=2
+      case default
+        j=3
+    end select
+
+
+    month = 'jan'
+    ! Condition can be integer,logical or character.
```
Comment (vendethiel) [inline fortran95.html.markdown]: Logical => booleans

Comment (vendethiel) [inline fortran95.html.markdown]: Also those are strings not characters

File: fortran95.html.markdown [hunk 7]
```diff
@@ -0,0 +1,445 @@
...
+      innerloop: do j=1,3    !loops can be named too.
+        exit    !quits the loop.
+      end do innerloop
+    cycle    !jump to next loop iteration.
+    enddo
+
+    
+    ! Goto statement exists but it is heavily discouraged though.
+    goto 10    
+    stop 1000   !stops code immediately (returning specified condition code).
```
Comment (vendethiel) [inline fortran95.html.markdown]: What's the 1000 condition code?

File: fortran95.html.markdown [hunk 8]
```diff
@@ -0,0 +1,445 @@
...
+10  j = 201    !this line is labeled as line 10
+    
+    
+    ! Arrays
+    ! ======
+    array = [1,2,3,4,5,6]
+    arrayb = [10.2,3e3,0.41,4e-5]
+    array2d =  reshape([1.0,2.0,3.0,4.0,5.0,6.0], [3,2])
+    
+    ! Fortran counts from 1
```
Comment (vendethiel) [inline fortran95.html.markdown]: "Fortran array indexing starts at 1"

File: fortran95.html.markdown (hunk 6 above)
Comment (vendethiel) [inline fortran95.html.markdown]: Ah, logical is fortran-speak, my bad

File: fortran95.html.markdown [hunk 9]
```diff
@@ -0,0 +1,445 @@
...
+    ! (better in general to put functions/procedures in the 'contains' section).
+    interface
+        elemental real function func4(a) result(res)
+            real, intent(in) :: a
+        end function func4
+    end interface
+    
+    ! Overloaded functions can be defined using named interfaces.
+    interface myabs
+    ! Can use 'module procedure' keyword to include functions already defined
```
Comment (vendethiel) [inline fortran95.html.markdown]: This should be intended

File: fortran95.html.markdown (hunk 1 above)
Comment (robochat) [inline fortran95.html.markdown]: will fix

File: fortran95.html.markdown (hunk 2 above)
Comment (robochat) [inline fortran95.html.markdown]: Do you mean on line 45? Does this really matter? In real code consistency is useful but in a tutorial showing variety demonstrates what the compiler will accept.

File: fortran95.html.markdown (hunk 3 above)
Comment (robochat) [inline fortran95.html.markdown]: You think that there are too many comments here? I will move line 26 to later in the tutorial.

File: fortran95.html.markdown (hunk 4 above)
Comment (robochat) [inline fortran95.html.markdown]: I wanted to introduce it early without going into (in my opinion) the distracting details of the print statement, too early in the tutorial. But I agree that it isn't very logical to put it here.

File: fortran95.html.markdown (hunk 5 above)
Comment (robochat) [inline fortran95.html.markdown]: will fix.

File: fortran95.html.markdown (hunk 7 above)
Comment (robochat) [inline fortran95.html.markdown]: The condition code is an integer so I just chose 1000 at random. Maybe there's more to it than that, I'll check.

File: fortran95.html.markdown (hunk 8 above)
Comment (robochat) [inline fortran95.html.markdown]: ok

File: fortran95.html.markdown (hunk 2 above)
Comment (vendethiel) [inline fortran95.html.markdown]: I mean `i, j, k = 1, m`

File: fortran95.html.markdown (hunk 7 above)
Comment (vendethiel) [inline fortran95.html.markdown]: If it's the process exit code you'll want to keep it under 256

File: fortran95.html.markdown (hunk 2 above)
Comment (vendethiel) [inline fortran95.html.markdown]: I think it's the only comment I have left before merging. It looks cleaner with correct spacing, methinks

Comment (robochat) [inline fortran95.html.markdown]: Ok, I've changed the spacing.

**Comment by vendethiel** [comment]
Good for me. Anything you want to address before I merge?:)

**Comment by robochat** [comment]
I just added a few examples of implied-Do loops to the array section. I think that I'm quite happy with it now.

**Comment by vendethiel** [comment]
Thanks! :cake:
